### PR TITLE
fix pytorch case found in hpc images

### PR DESCRIPTION
**Pull request by LiliDeng**

### CODE REVIEW

File: lisa/node.py [hunk 1]
```diff
@@ -394,6 +395,9 @@ def find_partition_with_freespace(
                     else:
                         mountpoint = partition.mountpoint
 
+                    if df.get_filesystem_available_space(mountpoint, True) < size_in_gb:
```
Comment (squirrelsc) [inline lisa/node.py]: There is a check `if not partition.size_in_gb >= size_in_gb:` already, why need to check here again?

File: lisa/node.py [hunk 2]
```diff
@@ -412,6 +416,9 @@ def find_partition_with_freespace(
                 else:
                     mountpoint = disk.mountpoint
 
+                if df.get_filesystem_available_space(mountpoint, True) < size_in_gb:
```
Comment (squirrelsc) [inline lisa/node.py]: What's the reason to check it again?

File: microsoft/testsuites/gpu/gpusuite.py [hunk 3]
```diff
@@ -45,6 +45,10 @@ class GpuTestSuite(TestSuite):
     TIMEOUT = 2000
 
     _pytorch_pattern = re.compile(r"^gpu count: (?P<count>\d+)", re.M)
+    _pytorch_error_pattern = re.compile(
```
Comment (squirrelsc) [inline microsoft/testsuites/gpu/gpusuite.py]: Name it like `numpy_error`

File: lisa/node.py (hunk 1 above)
Comment (LiliDeng) [reply lisa/node.py]: this is to compare total size, not free size,  I will remove this, just check free size

Comment (LiliDeng) [reply lisa/node.py]: updated

File: lisa/node.py (hunk 2 above)
Comment (LiliDeng) [reply lisa/node.py]: updated

File: microsoft/testsuites/gpu/gpusuite.py (hunk 3 above)
Comment (LiliDeng) [reply microsoft/testsuites/gpu/gpusuite.py]: updated

File: lisa/node.py (hunk 2 above)
Comment (squirrelsc) [reply lisa/node.py]: Move the logic to be shared. The return part happens on both `if/else`, so it can be moved out from `if/else` too.

Comment (LiliDeng) [reply lisa/node.py]: updated

**Review by squirrelsc — APPROVED** [review]

**Pull request merged**